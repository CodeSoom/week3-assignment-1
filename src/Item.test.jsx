import { render, fireEvent } from '@testing-library/react';

import Item from './Item';

describe('Item 컴포넌트 관련 테스트', () => {
  it('Item', () => {
    const task = {
      id: 1,
      title: '뭐라도 하기',
    };

    const handleClick = jest.fn();

    const { container, getByText } = render((
      <Item
        task={task}
        onClickDelete={handleClick}
      />
    ));

    expect(container).toHaveTextContent('뭐라도 하기');
    expect(container).toHaveTextContent('완료');

    expect(handleClick).not.toBeCalled();

    fireEvent.click(getByText('완료'));

    expect(handleClick).toBeCalledWith(1);
  });
});
