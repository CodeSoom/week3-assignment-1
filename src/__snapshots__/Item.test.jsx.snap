// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Item 1`] = `
<div>
  <li>
    뭐라도 하기
    <button
      type="button"
    >
      완료
    </button>
  </li>
</div>
`;
